momepy/preprocessing.py

```diff
@@ -1101,3 +1105,464 @@ def roundabout_simplification(
+    buffer : float
+        distance to buffer consolidated nodes in the Spider-web reconstruction
+    edge_from_col: string
+        edge attribute with the valid origin node id
+    edge_to_col: string
+        edge attribute with the valid destination node id
+
+    Returns
+    ----------
+    List
+        list of edges that should be added to the network. Edges are in the format
+        (origin_id, destination_id, data), where data is inferred from edges_gdf
+
+    """
+    # Determine what endpoints were made into clusters:
+    edges_gdf["origin_cluster"] = edges_gdf[edge_from_col].apply(
+        lambda u: nodes_dict[u] if u in nodes_dict else -1
+    )
+    edges_gdf["destination_cluster"] = edges_gdf[edge_to_col].apply(
+        lambda v: nodes_dict[v] if v in nodes_dict else -1
+    )
+
+    # Determine what edges need to be simplified (either between diff.
+    #  clusters or self-loops in a cluster):
+    edges_tosimplify_gdf = edges_gdf.query(
+        f"origin_cluster != destination_cluster or (('{edge_to_col}' == '{edge_from_col}') and origin_cluster >= 0)"
+    )
+
+    # Determine the new point geometries (when exists):
+    edges_tosimplify_gdf = edges_tosimplify_gdf.assign(
+        new_origin_pt=edges_tosimplify_gdf.origin_cluster.map(
+            cluster_gdf.geometry, None
+        )
+    )
+    edges_tosimplify_gdf = edges_tosimplify_gdf.assign(
+        new_destination_pt=edges_tosimplify_gdf.destination_cluster.map(
+            cluster_gdf.geometry, None
+        )
+    )
+
+    # Determine the new geometry according to the simplification method:
+    if method == "extend":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _extension_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    elif method == "euclidean":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _euclidean_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    elif method == "spider":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _spider_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt, buffer
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    else:
+        warnings.warn(
+            f"Simplification method '{method}' not recognized. Using spider-web simplification."
+        )
```

Comment:
I don't think this is a good behaviour. We should raise an error if the method is not implemented.

Reply:
threw a `ValueError` in that case